### ISSUE TITLE
ZeroDivisionError when parsing string "DECEMBER 21 19.87 87"

### ISSUE THREAD

**Issue by githubmlai** [scrapinghub/dateparser#401]
Hi,

I am getting the below zero division error when running the following code:

```
from dateparser.search import search_dates
s = "DECEMBER 21 19.87 87"
out = search_dates(s)
```
Environment:
```
dateparser==0.7.0
python==3.5.2
```

Exception:
```
Traceback (most recent call last):
 
  File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/search/__init__.py", line 33, in search_dates
    result = _search_with_detection.search_dates(text=text, languages=languages, settings=settings)
  


File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/conf.py", line 81, in wrapper
    return f(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/search/search.py", line 221, in search_dates
    settings=settings)}
  File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/search/search.py", line 153, in search_parse
    original=original, translated=translated, settings=settings)
  File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/search/search.py", line 140, in parse_found_objects
    parsed_best, substrings_best = self.choose_best_split(possible_parsed, possible_substrings)
  File "/home/ubuntu/.virtualenvs/dsh_v2/lib/python3.5/site-packages/dateparser/search/search.py", line 59, in choose_best_split
    rating.append([num_substrings, float(not_parsed)/float(num_substrings),
ZeroDivisionError: float division by zero
```

**Comment by emramos** [comment]
Same here:
`search_dates("a Ameri")`

`  File "search.py", line 59, in choose_best_split
    rating.append([num_substrings, float(not_parsed)/float(num_substrings),
ZeroDivisionError: float division by zero`

**Comment by lopuhin** [comment]
I think this is fixed by #429

**Comment by soferio** [comment]
Same issue here with:

>>> from dateparser.search import search_dates
>>> search_dates("National American Woman Suffrage Association")
